src/pybind/mgr/cephadm/module.py

```diff
@@ -2314,25 +2331,32 @@ def _calc_daemon_deps(self,
             root_cert = ''
             server_port = ''
             try:
-                server_port = str(self.cherrypy_thread.server_port)
-                root_cert = self.cherrypy_thread.ssl_certs.get_root_cert()
+                server_port = str(self.http_server.agent.server_port)
+                root_cert = self.http_server.agent.ssl_certs.get_root_cert()
             except Exception:
                 pass
             deps = sorted([self.get_mgr_ip(), server_port, root_cert,
                            str(self.device_enhanced_scan)])
         elif daemon_type == 'iscsi':
             deps = [self.get_mgr_ip()]
+
+        elif daemon_type == 'prometheus':
+            # for prometheus we add the active mgr as an explicit dependency,
+            # this way we force a redeploy after a mgr failover
+            deps.append(self.get_active_mgr().name())
+            deps.append(str(self.get_module_option_ex('prometheus', 'server_port', 9283)))
```

Comment:
I remember we talked about this, but defining the default, when a default is provided in the `MODULE_OPTION` is still weird... Isn't there any chance to get this port number from the source (maybe importing that from `mgr/prometheus`)?

If not, at least I'd use `self.DEFAULT_MGR_PROMETHEUS_PORT`, which is how this is done in other parts of this file (let's not hardcode literals across the code)...